maro/simulator/scenarios/vm_scheduling/business_engine.py

```diff
@@ -304,18 +349,48 @@ def _update_pm_workload(self):
                 vm = self._live_vms[vm_id]
                 total_pm_cpu_cores_used += vm.cpu_utilization * vm.cpu_cores_requirement
             pm.update_cpu_utilization(vm=None, cpu_utilization=total_pm_cpu_cores_used / pm.cpu_cores_capacity)
-            pm.energy_consumption = self._cpu_utilization_to_energy_consumption(cpu_utilization=pm.cpu_utilization)
+            pm.energy_consumption = self._cpu_utilization_to_energy_consumption(
+                sku=self._sku_dict[pm.sku_type],
+                cpu_utilization=pm.cpu_utilization
+            )
+
+    def _overload(self, pm_id: int):
+        """Overload logic.
+
+        Currently only support killing all VMs on the overload PM and note them as failed allocations.
+        """
+        # TODO: Future features of overload modeling.
+        #       1. Performance degradation
+        #       2. Quiesce specific VMs.
+        pm: PhysicalMachine = self._machines[pm_id]
+        vm_ids: List[int] = [vm_id for vm_id in pm.live_vms]
+
+        if self._kill_all_vms_if_overload:
+            for vm_id in vm_ids:
+                self._live_vms.pop(vm_id)
+
+            pm.deallocate_vms(vm_ids=vm_ids)
+            pm.update_cpu_utilization(vm=None, cpu_utilization=0)
```

Comment:
if want to record an energy consumption > 0, do not update utilization and re-calculate energy consumption here

Reply:
So in overload method, no matter it is enable `_kill_all_vms_if_overload` or not, it only counts the number of `overload_vms`.
And if it is enable `_kill_all_vms_if_overload` then deallocate vms. Let the next tick to calculate the workloads, right?

Reply:
yes